lib/manageiq/appliance_console/saml_authentication.rb

```diff
@@ -0,0 +1,212 @@
+        restart_httpd
+        true
+      rescue AwesomeSpawn::CommandResultError => e
+        log_command_error(e)
+        say("Failed to Unconfigure SAML Authentication - #{e}")
+        false
+      rescue => e
+        say("Failed to Unconfigure SAML Authentication - #{e}")
+        false
+      end
+
+      private
+
+      # Apache SAML Configuration
+
+      def rename_mellon_configfiles
+        debug_msg("Renaming mellon config files ...")
+        Dir.chdir(SAML2_CONFIG_DIRECTORY) do
+          Dir.glob("https_*.*") do |mellon_file|
+            saml2_file = nil
+            case mellon_file
+            when /^https_.*\.key$/  then saml2_file = "miqsp-key.key"
+            when /^https_.*\.cert$/ then saml2_file = "miqsp-cert.cert"
+            when /^https_.*\.xml$/  then saml2_file = "miqsp-metadata.xml"
+            end
+            if saml2_file
+              debug_msg("Renaming #{mellon_file} to #{saml2_file}")
+              File.rename(mellon_file, saml2_file)
+            end
+          end
+        end
+      end
+
+      def fetch_idp_metadata
+        idp_metadata = options[:saml_idp_metadata]
+        if path_is_file(idp_metadata) && idp_metadata != IDP_METADATA_FILE
+          debug_msg("Copying IDP metadata file #{idp_metadata} to #{IDP_METADATA_FILE} ...")
+          FileUtils.cp(idp_metadata, IDP_METADATA_FILE)
+        elsif path_is_url(idp_metadata)
+          debug_msg("Downloading IDP metadata file from #{idp_metadata}")
+          download_network_file(idp_metadata, IDP_METADATA_FILE)
+        end
+      end
+
+      def copy_apache_saml_configfiles
+        debug_msg("Copying Apache SAML Config files ...")
+        copy_template(HTTPD_CONFIG_DIRECTORY, "manageiq-remote-user.conf")
+        copy_template(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf")
+      end
+
+      def remove_apache_saml_configfiles
+        debug_msg("Removing Apache SAML Config files ...")
+        remove_file(HTTPD_CONFIG_DIRECTORY, "manageiq-remote-user.conf")
+        remove_file(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf")
+      end
+
+      def configured?
+        path_join(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf").exist?
+      end
+
+      def restart_httpd
+        httpd_service = LinuxAdmin::Service.new("httpd")
+        if httpd_service.running?
+          say("Restarting httpd ...")
+          httpd_service.restart
+        end
+      end
+
+      # SAML IDP Metadata
+
+      def validate_saml_idp_metadata_option
+        idp_metadata = options[:saml_idp_metadata]
+        raise "Must specify the SAML IDP metadata file or URL via --saml-idp-metadata" if idp_metadata.blank?
+
+        raise "Missing SAML IDP metadata file #{idp_metadata}" if path_is_file(idp_metadata) && !File.exist?(idp_metadata)
+      end
+
+      def path_is_file(path)
+        path.present? && !path_is_url(path)
+      end
+
+      def path_is_url(path)
+        path =~ /\A#{URI.regexp(["http", "https"])}\z/
```

Comment:
I'm not sure you can embed one regex inside of another like this

You may have to do...

```ruby
/\A#{URI.regexp(["http", "https"])}\z/x
```

or

```ruby
/\A#{Regexp.escape(URI.regexp(["http", "https"]))}\z/
```

Either that, you can just manually verify that you are anchored at the first position by checking that the match location is `0`